HttpCalls.playground/section-1.swift

```diff
@@ -33,18 +33,46 @@ func handleAddMatchResponse(data: NSData?, error: NSError?) {
     fetchCurrentQueue()
 }
 
+
 func logGETResults(data: NSData?, error: NSError?) {
-    var jsonResult: [NSDictionary] = NSJSONSerialization.JSONObjectWithData(
+    var jsonResult = convertToJSON(data)
+    jsonResult[jsonResult.count - 1].objectForKey("id")
+    jsonResult[jsonResult.count - 1].objectForKey("playerOne")
+    jsonResult[jsonResult.count - 1].objectForKey("playerTwo")
+}
+
+
+
+func deleteMatch() {
+    var responseData: NSData?
+    var request = NSMutableURLRequest(URL: NSURL(string: "http://localhost:4567/current_queue")!)
+    request.HTTPMethod = "GET"
+
+    NSURLConnection.sendAsynchronousRequest(
+        request,
+        queue: NSOperationQueue.mainQueue(),
+        completionHandler: {(response, data, error) in
+            var jsonResult = convertToJSON(data)
+            var matchId: AnyObject? = jsonResult[jsonResult.count - 1].objectForKey("id")
+
+            var deleteRequest = NSMutableURLRequest(URL: NSURL(string: "http://localhost:4567/delete_match?id=\(matchId as! String)")!);
+            deleteRequest.HTTPMethod = "DELETE"
+
+            NSURLConnection.sendAsynchronousRequest(deleteRequest, queue: NSOperationQueue.mainQueue(), completionHandler: {(response, data, error) -> Void in})
+
+        }
+    )
+}
+
+func convertToJSON(data: NSData?) -> [NSDictionary] {
+    return NSJSONSerialization.JSONObjectWithData(
         data!,
         options: NSJSONReadingOptions.MutableContainers,
         error: nil
         ) as! [NSDictionary]
-    jsonResult.count
-    jsonResult[jsonResult.count - 1].objectForKey("id")
-    jsonResult[jsonResult.count - 1].objectForKey("playerOne")
-    jsonResult[jsonResult.count - 1].objectForKey("playerTwo")
 }
 
 addMatch()
+deleteMatch()
```

Comment:
I went ahead and merged this, but the only thing I might change is to do this in the callback for `addMatch`, since right now, the delete could happen before add finishes.